test/e2e/node/node_problem_detector.go

```diff
@@ -390,7 +364,22 @@ func getNpdPodStat(ctx context.Context, f *framework.Framework, nodeName string)
 		break
 	}
 	if !hasNpdPod {
-		framework.Failf("No node-problem-detector pod is present in %+v", summary.Pods)
+		// on large clusters, some nodes return incomplete Summary of Pods :(
+		framework.Logf("unable to gather metrics from node %s, summary of pods on the node:%+v", nodeName, summary.Pods)
```

Comment:
@wojtek-t @hakuna-matatah What do you think of this change?

fyi, we can't run scale test for GCE on kops till this test is rewritten

Reply:
why do we have this dependency on this test?